tutorials/pages/shimmerevm-hardhat.md

```diff
@@ -0,0 +1,222 @@
+
+        require(block.timestamp >= unlockTime, "You can't withdraw yet");
+        require(msg.sender == owner, "You aren't the owner");
+
+        emit Withdrawal(address(this).balance, block.timestamp);
+
+        owner.transfer(address(this).balance);
+    }
+}
+```
+
+The above is an auto-generated smart contract by hardhat. You can replace it with your own smart contract. If you want to understand what the smart contract does, it's a Lock contract, which will lock funds for a specific time period and allows withdrawals only past that point.
+Some important things to note are data types. `event` means that we can subscribe to it like a webhook and wait for a ping to the client side whenever a withdrawal is triggered with `emit` from inside the function `withdraw()`.
+
+## Set Up the Contract
+
+1. Go to `hardhat.config.js`
+2. Update the `hardhat-config` with shimmer network details as shown below:
+
+```js
+module.exports = {
+  solidity: '0.8.18',
+  defaultNetwork: 'shimmerevm-testnet',
+  networks: {
+    'shimmerevm-testnet': {
+      url: 'https://json-rpc.evm.testnet.shimmer.network',
+      chainId: 1071,
+      accounts: [priv_key],
+    },
+  },
+};
+```
+
+:::info Private Key
+
+How to export the MetaMask private key of an account?
+
+- Click on the logo in the upper right corner
+
+- Select the account you want to export
+
+- On the account page, click the menu (three dots) in the upper right corner, then click the "Account Details" button:
+
+- Click "Export Private Key"
+
+- To access your private key, you will now need to enter your wallet password. When done, click Confirm to continue.
+
+- Your private key will now be displayed. Click to copy it and save it in a safe place. (Note: For obvious reasons, we didn't show it in the screenshots below - but yours will be there.)
+
+- Click close icon on top to close the screen.
+
+Follow [this article](https://helpwithpenny.com/export-and-import-metamask-private-key/) for further detail.
+
+:::
+
+1. Create a `.env` file in the project root to store your network id and jsonrpc if you want to keep it private. You can then refer to them by `process.env.SHIMMEREVM_TESTNET_JSONRPC`.
+
+:::note DOTENV
+
+Note that the file above requires DOTENV to manage environment variables and ethers and etherscan. Make sure to install all those packages.
+
+Find more instructions on how to use DOTENV in the [DOTENV npm package](https://www.npmjs.com/package/dotenv).
+
+:::
+
+## Compile the Contract
+
+To compile the contract, you first need to install the [Hardhat Toolbox package](https://www.npmjs.com/package/@nomicfoundation/hardhat-toolbox):
+
+```sh
+npm install --save-dev @nomicfoundation/hardhat-toolbox
+```
+
+Then, simply run to compile:
+
+```sh
+npx hardhat compile
+```
+
+## Deploy Your Contract on ShimmerEVM
+
+Run this command in root of the project directory:
+
+```sh
+npx hardhat run scripts/deploy.js --network shimmerevm-testnet
```

Comment:
Here I get:
`ProviderError: request might require more gas than it is allowed by the VM (50000000), or will never succeed`

Reply:
Do you've balance on your account?
It worked for me.

Reply:
Let me check if I used the correct private key. But maybe will need until tomorrow. It's late already 😅 
Sorry 😅

Reply:
My bad, I used the wrong private key with no funds on it. Thanks